src/components/common/UserProfileEdit/UserAdvancedSettings.tsx

```diff
@@ -26,20 +28,34 @@ const MSG = defineMessages({
     id: `${displayName}.heading`,
     defaultMessage: 'Advanced settings {learnMoreLink}',
   },
-  rpcErrorMessage: {
-    id: `${displayName}.rpcErrorMessage`,
+  invalidURLError: {
+    id: `${displayName}.invalidURLError`,
     defaultMessage: 'Enter a valid URL',
   },
+  invalidRPCError: {
+    id: `${displayName}.invalidRPCError`,
+    defaultMessage: 'Unable to validate RPC endpoint',
+  },
 });
 
 const validationSchema = object({
   [SlotKey.Metatransactions]: bool<boolean>(),
   [SlotKey.DecentralizedMode]: bool<boolean>(),
   [SlotKey.CustomRPC]: string()
     .defined()
-    .when(`${[SlotKey.DecentralizedMode]}`, {
+    .when(`${SlotKey.DecentralizedMode}`, {
       is: true,
-      then: string().url(() => MSG.rpcErrorMessage),
+      then: string()
+        .required(() => MSG.invalidURLError)
+        .url(() => MSG.invalidURLError)
+        .test(
+          'gnosisRpc',
+          () => MSG.invalidRPCError,
+          yupDebounce(validateCustomGnosisRPC, 200, {
+            isOptional: false,
+            circuitBreaker: isValidURL,
+          }),
```

Comment:
I'm good with this, at least for now. But we'll most likely have to change it prior to going live.

At the very least we'll need to be able to validate Ganache for dev mode _(which might have any chain id -- we can agree on a "proper" one)_.

Later, it will need to support basically every chain the CDapp will get deployed to

Reply:
Ok, opened an issue to keep track: https://github.com/JoinColony/colonyCDapp/issues/230